airstack/airstack.go

```diff
@@ -1,26 +1,123 @@
 package airstack
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
 	"fmt"
+	"io"
+	"net/http"
 	"strings"
+	"time"
 
 	ac "github.com/vocdoni/vote-frame/airstack/client"
+	"go.vocdoni.io/dvote/log"
 )
 
 // Airstack wraps all the required artifacts for interacting with the Airstack API
 type Airstack struct {
 	*ac.Client
+	maxHolders         uint32 // maxHolders is the maximum number of holders to be retrieved from the Airstack API
+	supportAPIEndpoint string // supportAPI is the URL of the support API
 }
 
 // NewAirstack creates a new Airstack artifact with a reference to a MongoDB and an Airstack client that
 // enables to make predefined queries to the Airstack GraphQL API.
-func NewAirstack(ctx context.Context, endpoint, apiKey, supportedBlockchains string) (*Airstack, error) {
+func NewAirstack(ctx context.Context, endpoint, apiKey, supportAPI, supportedBlockchains string, maxHolders uint32) (*Airstack, error) {
 	client, err := ac.NewClient(ctx, endpoint, apiKey, strings.Split(supportedBlockchains, ","))
 	if err != nil {
 		return nil, fmt.Errorf("error creating Airstack: %w", err)
 	}
 	return &Airstack{
-		Client: client,
+		Client:             client,
+		maxHolders:         maxHolders,
+		supportAPIEndpoint: supportAPI,
 	}, nil
 }
+
+func (a *Airstack) MaxHolders() uint32 {
+	return a.maxHolders
+}
+
+func (a *Airstack) NumHoldersByTokenAnkrAPI(tokenAddress, blockchain string) (uint32, error) {
+	if blockchain == "ethereum" {
+		blockchain = "eth" // Ankr API uses "eth" instead of "ethereum"
```

Comment:
and for the rest of the blockchains the name coincides?

Reply:
they are the same yes